src/engraving/libmscore/beam.cpp

```diff
@@ -1801,7 +1801,7 @@ void Beam::read(XmlReader& e)
             _userModified[idx] = true;
             double _spatium1 = spatium();
 
-            while (e.readNextStartElement()) {
+            while (e.readNextStartElement() && score()->mscVersion() >= 300) {
```

Comment:
Here, for < 3.0 scores, you do make the beam fragment userModified, but you don't set the py1/py2 values, so those will be zero. Doesn't that mean that the beam will always be flat, straddling the topmost staff line? Maybe better to not make it userModified in this case:
```c++
        } else if (tag == "Fragment") {
            if (score()->mscVersion() < 300) {
                e.skipCurrentElement();
            } else {
                BeamFragment* f = new BeamFragment;
                int idx = (_direction == DirectionV::AUTO || _direction == DirectionV::DOWN) ? 0 : 1;
                _userModified[idx] = true;
                double _spatium1 = spatium();

                while (e.readNextStartElement()) { 
                    ...

Reply:
That might also fix the vtests...